src/main/java/org/graylog2/gelfclient/GelfConfiguration.java

```diff
@@ -114,6 +117,15 @@ public InetSocketAddress getRemoteAddress() {
         return new InetSocketAddress(hostname, port);
     }
 
+    public URI getUri() {
+        return uri;
+    }
+
+    public GelfConfiguration uri(URI uri) {
+        this.uri = uri;
```

Comment:
The `uri` is set but it's not used in `#getRemoteAddress()`. We should probably check if the `uri` field is set and use it as an override for the `hostname` and `port` fields.

Reply:
The `getUri()` and `uri()` methods are also missing javadoc.